### PR TITLE
Org environment fix

### PR DESCRIPTION
**Pull request by sean-m-sullivan**
# What does this PR do?
bugfixes:
  - Fixed issue with organization role not acceppting default environments option correctly.
  - Fixed issue with licence role not operating properly, when a controller never had credentials provided for subscription lookup. See Role Readme for proper usuage.

# Is there a relevant Issue open for this?
#671 
#670 

# Other Relevant info, PRs, etc

#667 

@willtome 
@jrigglem

### CODE REVIEW

File: roles/license/README.md [hunk 1]
```diff
@@ -59,23 +61,20 @@ The module and this role can use either a manifest file, or lookup the subscript
...
+|`redhat_subscription_password`|""|yes|str|Red Hat or Red Hat Satellite password to get available subscriptions. Global var not part of controller_license data structure.|
 |`filters`|"default values"|no|str|dict of filters to use to narrow the subscription. See example below for how to use this.|
 |`support_level`|"Self-Support"|no|str|DEPRECATED - changed to `manifest_file` (still works as an alias)|
 |`list_num`|0|no|int|List index of the subscription to use, if you want to overide the default, it is recomended to use the filters to limit the pools found.|
-|`pool_id`|""|no|str|Red Hat or Red Hat Satellite pool_id to attach to, setting this will skip the lookup.|
+|`pool_id`|""|no|str|Red Hat or Red Hat Satellite pool_id to attach to.|
 |`force`|`False`|no|bool|By default, the license will only be applied if controller is currently unlicensed or trial licensed. When force=true, the license is always applied.|
+|`controller_license_use_lookup`|`False`|no|bool|Whether or not to lookup subscriptions. Global var not part of controller_license data structure.|
```
Comment (Tompage1994) [inline roles/license/README.md]: This shouldn't be in this bit of the variable listing but should instead be further up with the rest of the global vars

File: roles/license/README.md [hunk 2]
```diff
@@ -59,23 +61,20 @@ The module and this role can use either a manifest file, or lookup the subscript
...
 
-For further details on fields see <https://docs.ansible.com/automation-controller/latest/html/userguide/credential_plugins.html>
-
 ### License Variables for using Red Hat Subscription
 
 |Variable Name|Default Value|Required|Type|Description|
 |:---:|:---:|:---:|:---:|:---:|
-|`redhat_subscription_username`|""|no|str|Red Hat or Red Hat Satellite username to get available subscriptions.|
-|`redhat_subscription_password`|""|no|str|Red Hat or Red Hat Satellite password to get available subscriptions.|
+|`redhat_subscription_username`|""|yes|str|Red Hat or Red Hat Satellite username to get available subscriptions. Global var not part of controller_license data structure.|
```
Comment (Tompage1994) [inline roles/license/README.md]: Likewise, these should be above with the rest of the global vars

Comment (Tompage1994) [reply roles/license/README.md]: Also, I'm pretty certain these variables are not required if using a manifest. So they should be set as required: false

Comment (Tompage1994) [reply roles/license/README.md]: Can you also add any global vars to the `meta/argument_spec.yml` file for this role please

**Review by Tompage1994 — CHANGES_REQUESTED** [review]

File: roles/license/README.md (hunk 2 above)
Comment (sean-m-sullivan) [reply roles/license/README.md]: So there is no global section above, There are two sections used

Manifest
Subscription

so put each one in each, and in the subscription section, and maybe we should break it out as two seperate roles? or how should we structure the readme better. I have updated the argspec

**Review by Tompage1994 — APPROVED** [review]
LGTM

File: roles/license/defaults/main.yml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ---
 controller_configuration_license_secure_logging: "{{ controller_configuration_secure_logging | default('false') }}"
+controller_license_use_lookup: false
```
Comment (Tompage1994) [inline roles/license/defaults/main.yml]: Don't think this is needed actually?

File: changelogs/fragments/licence_org.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (Tompage1994) [inline changelogs/fragments/licence_org.yml:1]: We probably need a minor_change around the `use_lookup` change now

**Review by Tompage1994 — CHANGES_REQUESTED** [review]
Very minor points

**Review by silvinux — COMMENTED** [review]
LGTM!

**Review by Tompage1994 — APPROVED** [review]

**Review by djdanielsson — APPROVED** [review]
LGTM

**Comment by Tompage1994** [comment]
CI broken. Once #674 is merged this can be rebased and tested

**Pull request merged**